### PR TITLE
Fix #148 and AppManager API tests

### LINKED ISSUE #148: [BUG] Application events do not return UnsubscribeFunctions
**Describe the bug**
Unlike the appManager API events the Application events do not return UnsubscribeFunctions.

**To Reproduce**
See the @glue42/web typings.

**Expected behavior**
Like the appManager API events the Application events need to return UnsubscribeFunctions.

**Screenshots**
N/A

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Additional context**
The same applies to @glue42/desktop.

### PR DESCRIPTION
**Pull request by ksgeorgieva**
Fix #148 issue related to application events not returning unsubscribe function. 

Add validation for the type of the passed argument of application events methods.

Update of the **AppManager API** tests

### CODE REVIEW

File: e2e/tests/appManager/API/onInstanceStarted.spec.js [hunk 1]
```diff
@@ -1,110 +1,152 @@
...
+    afterEach(() => {
+        gtf.clearWindowActiveHooks();
+
+        return Promise.all(glue.appManager.instances().map((inst) => inst.stop()));
+    });
+
+    [undefined, null, '', false, 42, 'test'].forEach((invalidArg) => {
+        it(`Should throw an error when the passed argument (${JSON.stringify(invalidArg)}) isn\'t of type function`, () => {
+            try {
+                const un = glue.appManager.onInstanceStarted(invalidArg);
```
Comment (flashd2n) [inline e2e/tests/appManager/API/onInstanceStarted.spec.js:25]: false positive

File: e2e/tests/appManager/API/onInstanceStarted.spec.js [hunk 2]
```diff
@@ -1,110 +1,152 @@
...
+    it('Should return an unsubscribe function', () => {
+        const returned = glue.appManager.onInstanceStarted(() => {});
+
+        gtf.addWindowHook(returned);
+
+        expect(returned).to.be.a('function');
+    });
+
+    it('Should return a working unsubscribe function', (done) => {
+        const ready = gtf.waitFor(3, done);
```
Comment (flashd2n) [inline e2e/tests/appManager/API/onInstanceStarted.spec.js:64]: false positive

File: e2e/tests/appManager/API/onInstanceStopped.spec.js [hunk 3]
```diff
@@ -1,89 +1,158 @@
...
+    it('Should return an unsubscribe function', async () => {
+        const returned = glue.appManager.onInstanceStopped(() => {});
+        
+        gtf.addWindowHook(returned);
+
+        expect(returned).to.be.a("function");
+    });
+
+    it("Should return a working unsubscribe function", (done) => {
+        const ready = gtf.waitFor(3, done);
```
Comment (flashd2n) [inline e2e/tests/appManager/API/onInstanceStopped.spec.js:67]: false positive

File: e2e/tests/appManager/API/onInstanceStopped.spec.js [hunk 4]
```diff
@@ -1,89 +1,158 @@
...
+        });
+
+        Promise.all([app.start(), app.start(), app.start()])
+            .then(appInstArr => Promise.all(appInstArr.map(inst => inst.stop())))
+            .then(ready)
+            .catch(done);
+    });
+
+    it('Should replay by invoking the callback with the previous started instance', (done) => {
+        const ready = gtf.waitFor(2, done);
```
Comment (flashd2n) [inline e2e/tests/appManager/API/onInstanceStopped.spec.js:137]: Test in wrong spec

**Review by flashd2n — CHANGES_REQUESTED** [review]

File: e2e/tests/appManager/API/onAppChanged.spec.js [hunk 5]
```diff
@@ -1,106 +1,138 @@
...
+            try {
+                const un = glue.appManager.onAppChanged(invalidArg);
+                gtf.addWindowHook(un);
+                throw new Error('Should have thrown');
+            } catch (error) { }
+        });
+    });
+
+    it("Should invoke the callback when app definition is changed", (done) => {
+        const ready = gtf.waitFor(2, done);
```
[the remark this answers is not in the document]
Comment (ksgeorgieva) [reply e2e/tests/appManager/API/onAppChanged.spec.js:40]: There's an application definition with such name passed to the config object at the initialization of the Web Platform when starting GTF. See https://github.com/Glue42/core/blob/master/e2e/config/gtf/config.ts#L73  and https://github.com/Glue42/core/blob/master/e2e/config/gtf/index.ts#L95

The merged **extraDefOne** app definition changes the **url** property of the "Test" app and triggers **onAppChanged()**

**Review by flashd2n — CHANGES_REQUESTED** [review]

**Review by flashd2n — APPROVED** [review]

**Pull request merged**